03.bowling/frame.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+class Frame
+  attr_reader :shots
+
+  def initialize(shots, is_last_frame: false)
+    @shots = shots
+    @is_last_frame = is_last_frame
+  end
+
+  def pins
+    @shots.map(&:pin)
+  end
+
+  def score(index, all_frames)
+    base_score = @is_last_frame ? calc_base_score_for_last : calc_base_score
```

Comment:
```suggestion
    base_score = calc_base_score
```

として、最終フレームのボーナスは常に0を返すようにすると、いろいろとコードがシンプルになりそうです

Reply:
確かにそうですね、実際にかなりシンプルになりました！